vendor/diff-parsers-upstream-std.patch

```diff
@@ -0,0 +1,821 @@
+-          NEW PREFIXOP STRING TRUE UIDENT
++          NEW PREFIXOP STRING TRUE UIDENT UNDERSCORE
+           LBRACKETPERCENT QUOTED_STRING_EXPR
+ 
+ 
+ /* Entry points */
+ 
+@@ -1309,10 +1309,13 @@
+         { (* TODO review mkmod location *)
+           Pmod_apply(me1, mkmod ~loc:$sloc (Pmod_structure [])) }
+     | (* An extension. *)
+       ex = extension
+         { Pmod_extension ex }
++    | (* A hole. *)
++      UNDERSCORE
++        { Pmod_hole }
+     )
+     { $1 }
+ ;
+ 
+ (* A parenthesized module expression is a module expression that begins
+@@ -2273,12 +2276,14 @@
+   | indexop_expr(qualified_dotop, expr_semi_list, LESSMINUS v=expr {Some v})
+     { mk_indexop_expr user_indexing_operators ~loc:$sloc $1 }
+   | expr attribute
+       { Exp.attr $1 $2 }
+ /* BEGIN AVOID */
++  (*
+   | UNDERSCORE
+      { not_expecting $loc($1) "wildcard \"_\"" }
++  *)
+ /* END AVOID */
+ ;
+ %inline expr_attrs:
+   | LET MODULE ext_attributes mkrhs(module_name) module_binding_body IN seq_expr
+       { Pexp_letmodule($4, $5, $7), $3 }
+@@ -2312,14 +2317,10 @@
+       { Pexp_for($3, $5, $7, $6, $9), $2 }
+   | ASSERT ext_attributes simple_expr %prec below_HASH
+       { Pexp_assert $3, $2 }
+   | LAZY ext_attributes simple_expr %prec below_HASH
+       { Pexp_lazy $3, $2 }
+-  | OBJECT ext_attributes class_structure END
+-      { Pexp_object $3, $2 }
+-  | OBJECT ext_attributes class_structure error
+-      { unclosed "object" $loc($1) "end" $loc($4) }
+ ;
+ %inline expr_:
+   | simple_expr nonempty_llist(labeled_simple_expr)
+       { Pexp_apply($1, $2) }
+   | expr_comma_list %prec below_COMMA
+@@ -2368,10 +2369,14 @@
+       { Pexp_pack $4, $3 }
+   | LPAREN MODULE ext_attributes module_expr COLON package_type RPAREN
+       { Pexp_constraint (ghexp ~loc:$sloc (Pexp_pack $4), $6), $3 }
+   | LPAREN MODULE ext_attributes module_expr COLON error
+       { unclosed "(" $loc($1) ")" $loc($6) }
++  | OBJECT ext_attributes class_structure END
++      { Pexp_object $3, $2 }
++  | OBJECT ext_attributes class_structure error
++      { unclosed "object" $loc($1) "end" $loc($4) }
+ ;
+ %inline simple_expr_:
+   | mkrhs(val_longident)
+       { Pexp_ident ($1) }
+   | constant
+@@ -2403,10 +2408,12 @@
+       { Pexp_send($1, $3) }
+   | simple_expr op(HASHOP) simple_expr
+       { mkinfix $1 $2 $3 }
+   | extension
+       { Pexp_extension $1 }
++  | UNDERSCORE
++      { Pexp_hole }
+   | od=open_dot_declaration DOT mkrhs(LPAREN RPAREN {Lident "()"})
+       { Pexp_open(od, mkexp ~loc:($loc($3)) (Pexp_construct($3, None))) }
+   | mod_longident DOT LPAREN seq_expr error
+       { unclosed "(" $loc($3) ")" $loc($5) }
+   | LBRACE record_expr_content RBRACE
+@@ -2465,10 +2472,13 @@
+   | LABEL simple_expr %prec below_HASH
+       { (Labelled $1, $2) }
+   | TILDE label = LIDENT
+       { let loc = $loc(label) in
+         (Labelled label, mkexpvar ~loc label) }
++  | TILDE LPAREN label = LIDENT ty = type_constraint RPAREN
```

Comment:
Why this diff ?

Reply:
Due to https://github.com/ocaml-ppx/ocamlformat/pull/1756